cms/plugin_base.py

```diff
@@ -170,6 +170,16 @@ def render(self, context, instance, placeholder):
         return context
 
     @classmethod
+    def requires_parent_plugin(cls, slot, page):
+        requires_parent = cls.get_require_parent(slot, page)
+
+        if requires_parent:
+            return True
+
+        allowed_parents = cls().get_parent_classes(slot, page) or []
+        return bool(allowed_parents)
+
```

Comment:
Just a small suggestion: I don't see what the introduction of extra vars serves, the following is still quite readable.
````
def requires_parent_plugin(cls, slot, page):
    if cls.get_require_parent(slot, page):
        return True
    return bool(cls().get_parent_classes(slot, page))
````